packages/consoledot-api/src/queries/useSSOProvider.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from "@tanstack/react-query";
+import { useApiConfiguration } from "../ApiProvider";
+import { masQueries } from "../queryKeys";
+
+export type SsoProvider = {
+  tokenEndPointUrl: string;
+};
+
+export const useSSOProvider = () => {
+  const { basePath } = useApiConfiguration();
+
+  const { data: ssoProviders } = useQuery<SsoProvider>({
+    queryKey: masQueries.tokenEndPointUrl(),
+    queryFn: async () => {
+      const response = await fetch(
+        `${basePath}/api/kafkas_mgmt/v1/sso_providers`
+      );
+      const providers = await response.json();
+      const { token_url } = providers;
+      const ssoProvider: SsoProvider = { tokenEndPointUrl: token_url };
+      return ssoProvider;
+    },
+  });
+  const tokenEndPointUrl = ssoProviders?.tokenEndPointUrl;
+  return { tokenEndPointUrl };
+};
```

Comment:
```suggestion
export const useSSOTokenEndpoint = () => {
  const { basePath } = useApiConfiguration();

  return useQuery<string>({
    queryKey: masQueries.tokenEndPointUrl(),
    queryFn: async () => {
      const response = await fetch(
        `${basePath}/api/kafkas_mgmt/v1/sso_providers`
      );
      const provider = await response.json();
      const { token_url } = provider;
      if (token_url === undefined || token_url === '') {
        return Promise.reject("Invalid SSO token URL");
      }
      return token_url;
    },
  });
};
```

This is what I meant in my previous comments.